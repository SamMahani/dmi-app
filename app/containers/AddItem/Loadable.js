/**
 *
 * Asynchronously loads the component for AddItem
 *
 */

import loadable from 'loadable-components';

export default loadable(() => import('./index'));
