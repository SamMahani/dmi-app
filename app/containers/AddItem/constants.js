export const ADD_ITEM = 'app/AddItem/ADD_ITEM';
