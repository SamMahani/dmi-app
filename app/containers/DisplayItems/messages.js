/*
 * DisplayItems Messages
 *
 * This contains all the text for the DisplayItems container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.DisplayItems';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the DisplayItems container!',
  },
});
