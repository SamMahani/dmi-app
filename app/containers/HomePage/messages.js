/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.HomePage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Dovenmuehle Sample App',
  },
  body: {
    id: `${scope}.body`,
    defaultMessage: 'Body paragraph of the application.',
  },
});
