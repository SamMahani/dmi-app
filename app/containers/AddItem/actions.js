import { ADD_ITEM, ADD_ITEM_SUCCESS, ADD_ITEM_ERROR } from './constants';

export function addItem(item) {
  return {
    type: ADD_ITEM,
    item,
  };
}

export function addItemSuccess(success) {
  return {
    type: ADD_ITEM_SUCCESS,
    success,
  };
}

export function addItemError(error) {
  return {
    type: ADD_ITEM_ERROR,
    error,
  };
}
