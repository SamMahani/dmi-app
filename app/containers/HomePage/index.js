/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React, { Fragment } from 'react';
import { FormattedMessage } from 'react-intl';
import messages from './messages';
import Button from '../../components/UI/Button';
import Input from '../../components/Input/StyledInput';

/* eslint-disable react/prefer-stateless-function */
export default class HomePage extends React.PureComponent {
  handleClick() {
    console.log('Testing clicke');
  }

  render() {
    return (
      <Fragment>
        <h1>
          <FormattedMessage {...messages.header} />
        </h1>
        <p>
          <FormattedMessage {...messages.body} />
        </p>
        Add item:
        <Input />
        <Button type="submit" onClick={this.handleClick}>
          Submit
        </Button>
      </Fragment>
    );
  }
}
