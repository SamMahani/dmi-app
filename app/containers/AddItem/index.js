import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

// required components
import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import Form from '../../components/Form';
import Wrapper from '../../components/UI/Wrapper';
import Input from '../../components/Input';
import Button from '../../components/UI/Button';
import Heading from '../../components/UI/Heading';

import { itemSelector, successSelector, errorSelector } from './selectors';
import { addItem } from './actions';
import reducer from './reducer';
import saga from './saga';

class AddItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      input: '',
    };
    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  onSubmit(e) {
    e.preventDefault();
    const item = this.state.input;
    if (item) {
      this.props.addItem(item);
      this.setState({ input: '' });
    }
  }

  onChange(e) {
    const input = e.target.value;
    this.setState({ input });
  }

  render() {
    return (
      <Fragment>
        <Wrapper>
          <Form onSubmit={e => this.onSubmit(e)}>
            <Input
              inputType="text"
              label="Add item"
              labelColor="grey"
              onChange={this.onChange}
              value={this.state.input}
            />
            <Button type="submit" onClick={this.onSubmit}>
              Submit
            </Button>
          </Form>
        </Wrapper>
        <Wrapper>
          {this.props.success && <Heading>Post successful âœ…!</Heading>}
          {this.props.error && (
            <Heading background="#ff0000">Error when posting ðŸ“›</Heading>
          )}
        </Wrapper>
      </Fragment>
    );
  }
}

AddItem.propTypes = {
  addItem: PropTypes.func,
  success: PropTypes.bool,
  error: PropTypes.bool,
};

const mapStateToProps = createStructuredSelector({
  item: itemSelector(),
  success: successSelector(),
  error: errorSelector(),
});

function mapDispatchToProps(dispatch) {
  return {
    addItem: item => dispatch(addItem(item)),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'addItem', reducer });
const withSaga = injectSaga({ key: 'addItem', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(AddItem);
