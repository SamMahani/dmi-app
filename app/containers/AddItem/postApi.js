import { put } from 'redux-saga/effects';
import { addItemSuccess, addItemError } from './actions';

export function* postApi(action) {
  const { item } = action;
  try {
    const response = yield fetch('/api/addItem', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json; charset=utf-8',
      },
      body: JSON.stringify({ item }),
    });
    console.log(response);
    yield put(addItemSuccess());
  } catch (error) {
    yield put(addItemError());
  }
}
