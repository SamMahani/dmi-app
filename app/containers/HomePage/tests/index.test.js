import React from 'react';
import { FormattedMessage } from 'react-intl';
import { shallow } from 'enzyme';

import HomePage from '../index';
import AddItem from '../../AddItem';
import messages from '../messages';

describe('<HomePage />', () => {
  it('should render the page message', () => {
    const renderedComponent = shallow(<HomePage />);
    expect(
      renderedComponent.contains(<FormattedMessage {...messages.header} />),
    ).toEqual(true);
  });

  it('should render a Wrapper Component', () => {
    const renderedComponent = shallow(<HomePage />);
    expect(renderedComponent.find('Wrapper').length).toBe(1);
  });

  it('should render addItem component', () => {
    const renderedComponent = shallow(
      <HomePage location={{ pathname: '/' }} />,
    );
    expect(renderedComponent.contains(<AddItem />)).toEqual(true);
  });

  it('should match render of HomePage', () => {
    const renderedComponent = shallow(<HomePage />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should match render of addItem component', () => {
    const renderedComponent = shallow(<HomePage />);
    const addItemComponent = renderedComponent.find('AddItem');
    expect(addItemComponent).toMatchSnapshot();
  });
});
