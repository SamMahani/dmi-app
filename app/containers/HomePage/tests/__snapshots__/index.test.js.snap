// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should match render of HomePage 1`] = `
<Fragment>
  <Wrapper
    flexDirection="column"
  >
    <h1>
      <FormattedMessage
        defaultMessage="Dovenmuehle Sample App"
        id="app.containers.HomePage.header"
        values={Object {}}
      />
    </h1>
  </Wrapper>
  <withReducer(withSaga(Connect(AddItem))) />
</Fragment>
`;

exports[`<HomePage /> should match render of addItem component 1`] = `null`;
